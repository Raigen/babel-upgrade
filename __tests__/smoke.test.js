const JSON5 = require('json5');
const fs = require('fs');
const pify = require('pify');
const crossSpawn = require('cross-spawn');
const rimraf = require('rimraf');

async function gitCheckout(dir, repo, branch = 'master') {
  const gitArgs = ['clone', repo, '--branch', branch, '--single-branch', dir];

  // pify only works with callback style
  // crossSpawn which uses child_process.spawn has no callback style
  return new Promise((resolve, reject) => {
    const child = crossSpawn('git', gitArgs);
    child.on('error', (err) => reject(err, err.stack));
    child.on('close', () => resolve());
  });
}

async function babelUpgrade(dir) {
  // pify only works with callback style
  // crossSpawn which uses child_process.spawn has no callback style
  return new Promise((resolve, reject) => {
    const child = crossSpawn('node', ['../../bin/babel-upgrade'], { cwd: dir });
    child.on('error', (err) => reject(err, err.stack));
    child.on('close', () => resolve());
  });
}

async function gitDiff(dir) {
  // pify only works with callback style
  // crossSpawn which uses child_process.spawn has no callback style
  return new Promise((resolve, reject) => {
    let result = '';
    const child = crossSpawn('git', ['--no-pager', 'diff', '--no-color', '-U0'], { cwd: dir });
    child.stdout.on('data', d => result += d.toString());
    child.on('error', err => reject(err));
    child.on('close', () => resolve(result));
  })
}

describe('smoke tests', function () {
  afterAll(async function() {
    await pify(rimraf)('./filetests');
  });

  it('babel-preset-env', async function () {
    const dir = './filetests/babel-preset-env';

    await gitCheckout(dir, 'https://github.com/babel/babel-preset-env', '1.x');
    await babelUpgrade(dir);

    expect(await gitDiff(dir)).toMatchSnapshot();
  }, 10000);

  it('screeps-flowtype-jest-skeleton', async function () {
    const dir = './filetests/screeps-flowtype-jest-skeleton';

    await gitCheckout(dir, 'https://github.com/FossiFoo/screeps-flowtype-jest-skeleton');
    await babelUpgrade(dir);

    expect(await gitDiff(dir)).toMatchSnapshot();
  }, 10000);

  it('react-side-effect', async function () {
    const dir = './filetests/react-side-effect';

    await gitCheckout(dir, 'https://github.com/gaearon/react-side-effect');
    await babelUpgrade(dir);

    expect(await gitDiff(dir)).toMatchSnapshot();
  }, 10000);
});
