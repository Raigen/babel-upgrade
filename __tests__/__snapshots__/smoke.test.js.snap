// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`smoke tests babel-preset-env 1`] = `
"diff --git a/package.json b/package.json
index cafdc8a..9444a6c 100644
--- a/package.json
+++ b/package.json
@@ -22 +22 @@
-    \\"test-only\\": \\"mocha ./test --compilers js:babel-register -t 10000\\"
+    \\"test-only\\": \\"mocha ./test --compilers js:@babel/register -t 10000\\"
@@ -25 +25,25 @@
-    \\"babel-plugin-check-es2015-constants\\": \\"^6.22.0\\",
+    \\"@babel/core\\": \\"7.0.0-beta.40\\",
+    \\"@babel/plugin-transform-arrow-functions\\": \\"7.0.0-beta.40\\",
+    \\"@babel/plugin-transform-async-to-generator\\": \\"7.0.0-beta.40\\",
+    \\"@babel/plugin-transform-block-scoped-functions\\": \\"7.0.0-beta.40\\",
+    \\"@babel/plugin-transform-block-scoping\\": \\"7.0.0-beta.40\\",
+    \\"@babel/plugin-transform-classes\\": \\"7.0.0-beta.40\\",
+    \\"@babel/plugin-transform-computed-properties\\": \\"7.0.0-beta.40\\",
+    \\"@babel/plugin-transform-destructuring\\": \\"7.0.0-beta.40\\",
+    \\"@babel/plugin-transform-duplicate-keys\\": \\"7.0.0-beta.40\\",
+    \\"@babel/plugin-transform-exponentiation-operator\\": \\"7.0.0-beta.40\\",
+    \\"@babel/plugin-transform-for-of\\": \\"7.0.0-beta.40\\",
+    \\"@babel/plugin-transform-function-name\\": \\"7.0.0-beta.40\\",
+    \\"@babel/plugin-transform-literals\\": \\"7.0.0-beta.40\\",
+    \\"@babel/plugin-transform-modules-amd\\": \\"7.0.0-beta.40\\",
+    \\"@babel/plugin-transform-modules-commonjs\\": \\"7.0.0-beta.40\\",
+    \\"@babel/plugin-transform-modules-systemjs\\": \\"7.0.0-beta.40\\",
+    \\"@babel/plugin-transform-modules-umd\\": \\"7.0.0-beta.40\\",
+    \\"@babel/plugin-transform-object-super\\": \\"7.0.0-beta.40\\",
+    \\"@babel/plugin-transform-parameters\\": \\"7.0.0-beta.40\\",
+    \\"@babel/plugin-transform-regenerator\\": \\"7.0.0-beta.40\\",
+    \\"@babel/plugin-transform-shorthand-properties\\": \\"7.0.0-beta.40\\",
+    \\"@babel/plugin-transform-spread\\": \\"7.0.0-beta.40\\",
+    \\"@babel/plugin-transform-sticky-regex\\": \\"7.0.0-beta.40\\",
+    \\"@babel/plugin-transform-template-literals\\": \\"7.0.0-beta.40\\",
+    \\"@babel/plugin-transform-unicode-regex\\": \\"7.0.0-beta.40\\",
@@ -27,21 +50,0 @@
-    \\"babel-plugin-transform-async-to-generator\\": \\"^6.22.0\\",
-    \\"babel-plugin-transform-es2015-arrow-functions\\": \\"^6.22.0\\",
-    \\"babel-plugin-transform-es2015-block-scoped-functions\\": \\"^6.22.0\\",
-    \\"babel-plugin-transform-es2015-block-scoping\\": \\"^6.23.0\\",
-    \\"babel-plugin-transform-es2015-classes\\": \\"^6.23.0\\",
-    \\"babel-plugin-transform-es2015-computed-properties\\": \\"^6.22.0\\",
-    \\"babel-plugin-transform-es2015-destructuring\\": \\"^6.23.0\\",
-    \\"babel-plugin-transform-es2015-duplicate-keys\\": \\"^6.22.0\\",
-    \\"babel-plugin-transform-es2015-for-of\\": \\"^6.23.0\\",
-    \\"babel-plugin-transform-es2015-function-name\\": \\"^6.22.0\\",
-    \\"babel-plugin-transform-es2015-literals\\": \\"^6.22.0\\",
-    \\"babel-plugin-transform-es2015-modules-amd\\": \\"^6.22.0\\",
-    \\"babel-plugin-transform-es2015-modules-commonjs\\": \\"^6.23.0\\",
-    \\"babel-plugin-transform-es2015-modules-systemjs\\": \\"^6.23.0\\",
-    \\"babel-plugin-transform-es2015-modules-umd\\": \\"^6.23.0\\",
-    \\"babel-plugin-transform-es2015-object-super\\": \\"^6.22.0\\",
-    \\"babel-plugin-transform-es2015-parameters\\": \\"^6.23.0\\",
-    \\"babel-plugin-transform-es2015-shorthand-properties\\": \\"^6.22.0\\",
-    \\"babel-plugin-transform-es2015-spread\\": \\"^6.22.0\\",
-    \\"babel-plugin-transform-es2015-sticky-regex\\": \\"^6.22.0\\",
-    \\"babel-plugin-transform-es2015-template-literals\\": \\"^6.22.0\\",
@@ -49,3 +51,0 @@
-    \\"babel-plugin-transform-es2015-unicode-regex\\": \\"^6.22.0\\",
-    \\"babel-plugin-transform-exponentiation-operator\\": \\"^6.22.0\\",
-    \\"babel-plugin-transform-regenerator\\": \\"^6.22.0\\",
@@ -57 +57,6 @@
-    \\"babel-cli\\": \\"^6.23.0\\",
+    \\"@babel/cli\\": \\"7.0.0-beta.40\\",
+    \\"@babel/core\\": \\"7.0.0-beta.40\\",
+    \\"@babel/helper-fixtures\\": \\"7.0.0-beta.40\\",
+    \\"@babel/helper-plugin-test-runner\\": \\"7.0.0-beta.40\\",
+    \\"@babel/preset-env\\": \\"7.0.0-beta.40\\",
+    \\"@babel/register\\": \\"7.0.0-beta.40\\",
@@ -59,2 +63,0 @@
-    \\"babel-helper-fixtures\\": \\"^6.22.0\\",
-    \\"babel-helper-plugin-test-runner\\": \\"^6.22.0\\",
@@ -62,2 +64,0 @@
-    \\"babel-preset-env\\": \\"^1.4.0\\",
-    \\"babel-register\\": \\"^6.23.0\\",
@@ -66,0 +68 @@
+    \\"electron-to-chromium\\": \\"^1.3.11\\",
@@ -74 +75,0 @@
-    \\"electron-to-chromium\\": \\"^1.3.11\\",
@@ -80 +81 @@
-        \\"env\\",
+        \\"@babel/preset-env\\",
"
`;

exports[`smoke tests react-side-effect 1`] = `
"diff --git a/.babelrc b/.babelrc
index f940c90..d018eea 100644
--- a/.babelrc
+++ b/.babelrc
@@ -3,2 +3,10 @@
-    \\"react\\",
-    [\\"env\\", { \\"targets\\": {\\"node\\": \\"current\\"}, \\"loose\\": true }]
+    \\"@babel/preset-react\\",
+    [
+      \\"@babel/preset-env\\",
+      {
+        \\"targets\\": {
+          \\"node\\": \\"current\\"
+        },
+        \\"loose\\": true
+      }
+    ]
@@ -7,2 +15,2 @@
-    \\"transform-object-rest-spread\\",
-    \\"transform-class-properties\\",
+    \\"@babel/plugin-proposal-object-rest-spread\\",
+    \\"@babel/plugin-proposal-class-properties\\",
diff --git a/package.json b/package.json
index ef1db4c..f7a3e80 100644
--- a/package.json
+++ b/package.json
@@ -41,2 +41,8 @@
-    \\"babel-cli\\": \\"^6.18.0\\",
-    \\"babel-core\\": \\"^6.21.0\\",
+    \\"@babel/cli\\": \\"7.0.0-beta.40\\",
+    \\"@babel/core\\": \\"7.0.0-beta.40\\",
+    \\"@babel/node\\": \\"7.0.0-beta.40\\",
+    \\"@babel/plugin-proposal-class-properties\\": \\"7.0.0-beta.40\\",
+    \\"@babel/plugin-proposal-object-rest-spread\\": \\"7.0.0-beta.40\\",
+    \\"@babel/preset-env\\": \\"7.0.0-beta.40\\",
+    \\"@babel/preset-react\\": \\"7.0.0-beta.40\\",
+    \\"@babel/register\\": \\"7.0.0-beta.40\\",
@@ -44,5 +49,0 @@
-    \\"babel-plugin-transform-class-properties\\": \\"^6.24.1\\",
-    \\"babel-plugin-transform-object-rest-spread\\": \\"^6.26.0\\",
-    \\"babel-preset-env\\": \\"^1.6.0\\",
-    \\"babel-preset-react\\": \\"^6.16.0\\",
-    \\"babel-register\\": \\"^6.18.0\\",
@@ -61 +62 @@
-    \\"rollup-plugin-babel\\": \\"^3.0.2\\",
+    \\"rollup-plugin-babel\\": \\"^4.0.0-beta.2\\",
diff --git a/test/mocha.opts b/test/mocha.opts
index cb6872d..f1cc49a 100644
--- a/test/mocha.opts
+++ b/test/mocha.opts
@@ -1 +1 @@
---require babel-register
+--require @babel/register
"
`;

exports[`smoke tests screeps-flowtype-jest-skeleton 1`] = `
"diff --git a/.babelrc b/.babelrc
index 5788e0a..2ae8bf8 100644
--- a/.babelrc
+++ b/.babelrc
@@ -2,2 +2,7 @@
-  \\"presets\\": [\\"es2015\\"],
-  \\"plugins\\": [\\"transform-flow-strip-types\\"]
+  \\"presets\\": [
+    \\"@babel/preset-env\\",
+    \\"@babel/preset-flow\\"
+  ],
+  \\"plugins\\": [
+    \\"@babel/plugin-transform-flow-strip-types\\"
+  ]
diff --git a/package.json b/package.json
index 867dd23..05afe19 100644
--- a/package.json
+++ b/package.json
@@ -6 +6,3 @@
-  \\"dependencies\\": {},
+  \\"dependencies\\": {
+    \\"@babel/preset-flow\\": \\"7.0.0-beta.40\\"
+  },
@@ -8 +10,5 @@
-    \\"babel-cli\\": \\"^6.18.0\\",
+    \\"@babel/cli\\": \\"7.0.0-beta.40\\",
+    \\"@babel/core\\": \\"7.0.0-beta.40\\",
+    \\"@babel/preset-env\\": \\"7.0.0-beta.40\\",
+    \\"@babel/preset-flow\\": \\"7.0.0-beta.40\\",
+    \\"babel-core\\": \\"^7.0.0-bridge.0\\",
@@ -11 +16,0 @@
-    \\"babel-preset-latest\\": \\"^6.16.0\\",
@@ -33 +38,3 @@
-    \\"setupFiles\\": [\\"./etc/jestsetup.js\\"],
+    \\"setupFiles\\": [
+      \\"./etc/jestsetup.js\\"
+    ],
"
`;
